src/connection/connection.rs

```diff
@@ -137,6 +130,10 @@ impl Connection {
         self.server_connections.values()
     }
 
+    pub fn username(&self) -> Option<String> {
+        self.username.clone()
+    }
```

Comment:
These kinds of getters shouldn't clone, but give the caller the option to. This should return `Option<&str>`, in fact.